code/game/gamemodes/objective.dm

```diff
@@ -176,14 +182,12 @@ GLOBAL_LIST_INIT(potential_theft_objectives, (subtypesof(/datum/theft_objective)
 	martyr_compatible = TRUE
 	var/won = FALSE
 
-/datum/objective/assassinateonce/find_target(list/target_blacklist)
-	..()
+/datum/objective/assassinateonce/update_explanation_text()
 	if(target?.current)
-		explanation_text = "Teach [target.current.real_name], the [target.assigned_role], a lesson they will not forget. The target only needs to die once for success."
+		explanation_text = "Teach [target.current.real_name], the [target.assigned_role == target.special_role ? (target.special_role) : (target.assigned_role)], a lesson they will not forget. The target only needs to die once for success."
```

Comment:
This will always return target.assigned_role, you should probably return all of these to just `target.assigned_role`. In another PR, we could tweak it so that if they have no assigned role, it shows the special role I guess